src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs

```diff
@@ -39,7 +39,15 @@ public async Task ExecuteAsync()
 			await Task.WhenAll(files.Select(file 
 				=> NavigationHelpers.OpenPath(file.path, _pageContext.ShellPage!)));
 
-			folders.ForEach(async folder => await NavigationHelpers.OpenPathInNewTab(folder.path, false));
+			var newTabsTasks = folders.Select(async folder =>
+			{
+				await NavigationHelpers.OpenPathInNewTab(folder.path, false);
+			});
+			
+			foreach (var task in newTabsTasks)
+			{
+				await task;
+			}
```

Comment:
Can you use `await Task.WhenAll(newTabsTasks)` instead?

Reply:
If it's ok to run out of order this, I can make the change.

Reply:
I think the order here does not matter, see 1137d1945ea8734c425cb44b6c5e3003b7b41dff

Reply:
Opening new tabs, I think should be based on the order of the items in the list, otherwise it can appear as a bug to the user when the same action sometimes has different results.

Reply:
Agree, changed to `foreach`